mkdocs/utils/__init__.py

```diff
@@ -431,7 +435,10 @@ def nest_paths(paths):
 
         branch = nested
         for part in parts:
-            part = filename_to_title(part)
+            dir = False
+            if os.path.basename(part) != path:
+                dir = True
```

Comment:
I'm not sure that works since filename_to_title is called later from `nav.py` to add files to the TOC when `_path_to_page` is called. I tried a few variations on what you described, but to be honest I think they just complicate things a bit more (at least the ways I tried).